### PR TITLE
[DS-4574] v. 7 - Upgrade DBCP2 dependency

### PR DESCRIPTION
**Pull request by J4bbi**
Upgrade DBCP2 to newest version, 2.8.0. Current version 2.1.1. is from 2015. JDBC 4.2 (Java 8) support from v. 2.5.0. Bugfixes and performance improvements. Could help with the slow performance reported with large repos using Hibernate (6.x)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -728,10 +735,16 @@
...
+                        <groupId>org.apache.commons</groupId>
+                        <artifactId>commons-dbcp2</artifactId>
+                        <version>2.8.0</version>
+                    </dependency>
                 </dependencies>
             </dependencyManagement>
         </profile>
-        <!-- If running Java 7, use JRE7 PostgreSQL driver -->
+
+        <!-- Java 7 -->
```
Comment (tdonohue) [inline pom.xml:747]: Minor point.  If you are willing, we can *remove* all this logic pertaining to Java 7 (and even remove mention of Java 8).  DSpace 7 only supports Java 11 (or above).  It seems we left around this old Java 7/8 profile stuff by accident.  (This isn't required for this PR though, just noting that this entire section can be removed -- we can always remove it later in a follow-up PR)

**Review by tdonohue — APPROVED** [review]
@J4bbi : Thanks for this.  Overall this looks good. I did have a minor inline comment about the Java 7 / 8 profiles here...they are no longer needed.  If you are willing to clean them up, this could be a good opportunity to just remove them.  Otherwise, we can clean them up in a future PR.

Overall, I'm +1 the changes here...but I'll leave this open until we determine whether to remove the Java 7 & 8 profiles

File: pom.xml (hunk 1 above)
Comment (J4bbi) [reply pom.xml]: I believe that's done. I removed the profiles and moved the postgres dependency to 3rd party dependencies.

I was a bit confused about there being a merge-conflict, first I re-pulled, rebased and resolved the conflict. Don't think I've broken anything ha ha

**Comment by mwoodiupui** [comment]
Should we also upgrade Commons Pool to 2.9?

File: pom.xml [hunk 2]
```diff
@@ -1070,6 +1034,20 @@
 
             <!-- DSpace third Party Dependencies -->
 
+            <dependency>
+                <groupId>org.postgresql</groupId>
+                <artifactId>postgresql</artifactId>
+                <version>${postgresql.driver.version}</version>
+            </dependency>
+
+            <!-- Solr introduces multiple versions of zookeeper. So,
```
Comment (mwoodiupui) [inline pom.xml:1043]: I don't understand why Zookeeper came in with this patch.

Comment (mwoodiupui) [reply pom.xml:1043]: Never mind; SolrJ wants it for some reason.

**Review by mwoodiupui — APPROVED** [review]
Looks well.
Built, installed, tested from the command line.  No obvious problems.

**Pull request merged**